src/WinRT.Runtime/Projections/ICustomPropertyProvider.net5.cs

```diff
@@ -23,6 +25,105 @@ interface ICustomProperty
+        /// </summary>
+        /// <param name="name">The name of the property to get.</param>
+        /// <returns>The <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation for the property specified by <paramref name="name"/>.</returns>
+        public Microsoft.UI.Xaml.Data.BindableCustomProperty GetProperty(string name);
+
+        /// <summary>
+        /// Get the generated <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation representing the specified index property type.
+        /// </summary>
+        /// <param name="indexParameterType">The index property to get.</param>
+        /// <returns>The <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation for the property specified by <paramref name="indexParameterType"/>.</returns>
+        public Microsoft.UI.Xaml.Data.BindableCustomProperty GetProperty(Type indexParameterType);
+    }
+
+    /// <summary>
+    /// An <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation that relies on a source generation approach for its implememtation
+    /// rather than reflection.  This is used by the source generator generating the implementation for <see cref="Microsoft.UI.Xaml.Data.IBindableCustomPropertyImplementation"/>.
+    /// </summary>
+    [global::WinRT.WinRTExposedType(typeof(global::ABI.Microsoft.UI.Xaml.Data.ManagedCustomPropertyWinRTTypeDetails))]
+#if EMBED
+    internal
+#else
+    public
+#endif
+    sealed class BindableCustomProperty : ICustomProperty
+    {
+        private readonly bool _canRead;
+        private readonly bool _canWrite;
+        private readonly string _name;
+        private readonly Type _type;
+        private readonly Func<object, object> _getValue;
+        private readonly Action<object, object> _setValue;
+        private readonly Func<object, object, object> _getIndexedValue;
+        private readonly Action<object, object, object> _setIndexedValue;
+
+        public BindableCustomProperty(
+            bool canRead,
+            bool canWrite,
+            string name,
+            Type type,
+            Func<object, object> getValue,
+            Action<object, object> setValue,
+            Func<object, object, object> getIndexedValue,
+            Action<object, object, object> setIndexedValue)
+        {
+            _canRead = canRead;
+            _canWrite = canWrite;
+            _name = name;
+            _type = type;
+            _getValue = getValue;
+            _setValue = setValue;
+            _getIndexedValue = getIndexedValue;
+            _setIndexedValue = setIndexedValue;
+        }
+
+        bool ICustomProperty.CanRead => _canRead;
+
+        bool ICustomProperty.CanWrite => _canWrite;
+
+        string ICustomProperty.Name => _name;
+
+        Type ICustomProperty.Type => _type;
+
+        object ICustomProperty.GetIndexedValue(object target, object index) => _getIndexedValue != null ? _getIndexedValue(target, index) : throw new NotImplementedException();

```

Comment:
Is `NotImplementedException` by design? Ie. should it not be `NotSupportedException` or some other exception?

Reply:
There are scenarios where these `NotImplementedException` can be seen as `NotSupportException`.  i.e. someone using the `indexer` function when it is a regular property.  But then there are scenarios where if the set accessor or get is not implemented by the user, maybe it should be `NotImplementedException` rather than `NotSupportedException`.  So, for consistency throughout and to avoid trying to figure out the actual intention, I think `NotImplementedException` probably makes sense to keep unless someone is looking for the exact hresult and there is a contract to maintain.